core/src/kura.rs

```diff
@@ -449,81 +391,17 @@ pub struct BlockCount {
     pub skip: usize,
 }
 
-/// Marker struct for typestate, signifies that store is protected by
-/// a lockfile owned by this instance of [`BlockStore`] and thus can be written into.
-///
-/// Owns the path to store because [`Unlocked`] needs
-/// to unlock it when dropped, and `Drop` impls
-/// cannot be specialized, so `Drop` has to be implemented
-/// on typestate struct.
-#[derive(Debug, Deref)]
-pub struct Locked(PathBuf);
-impl Lock for Locked {}
-
-impl Locked {
-    // Try to lift lockfile. Only for internal use in `unlock`
-    // and `Drop` implementation.
-    fn try_lift_lock(&mut self) -> Result<()> {
-        let path = self.join(LOCK_FILE_NAME);
-        if let Err(e) = fs::remove_file(path.clone()) {
-            match e.kind() {
-                std::io::ErrorKind::NotFound => Err(Error::Unlocked(self.0.clone())),
-                _ => Err(Error::IO(e, path)),
-            }
-        } else {
-            Ok(())
-        }
-    }
-
-    // Lift lockfile and convert self to [`Unlocked`]
-    fn unlock(mut self) -> Result<Unlocked> {
-        // Lift the lockfile manually
-        self.try_lift_lock()?;
-
-        // Prevent destructor from running (lockfile is already deleted)
-        let mut me = core::mem::ManuallyDrop::new(self);
-
-        let path = core::mem::take(&mut me.0);
-
-        Ok(Unlocked(path))
-    }
-}
-
-impl Drop for Locked {
-    fn drop(&mut self) {
-        if let Err(err) = self.try_lift_lock() {
-            warn!(%err, "Couldn't remove lockfile at {:?}. Either the lockfile was deleted externally while Iroha instance was running, or there was an IO error. Check the block store with kura_inspector and then delete the lockfile manually.", self);
-        }
-    }
-}
-
 /// An implementation of a block store for `Kura`
 /// that uses `std::fs`, the default IO file in Rust.
 #[derive(Debug)]
-pub struct BlockStore<L: Lock> {
-    path_to_blockchain: L,
+pub struct BlockStore {
+    path_to_blockchain: PathBuf,
 }
 
-/// Operations available only on unlocked store,
-/// i.e. creation and locking.
-impl BlockStore<Unlocked> {
-    /// Create a new read-only block store in `path`.
-    pub fn new(path: &Path) -> Self {
-        BlockStore {
-            path_to_blockchain: Unlocked::new(path.to_path_buf()),
-        }
-    }
-
-    /// Try to acquire a lockfile and convert self to
-    /// read-write [`BlockStore<Locked>`]
-    ///
-    /// # Errors
-    /// - IO errors
-    /// - lockfile already exists
-    pub fn lock(self) -> Result<BlockStore<Locked>> {
-        Ok(BlockStore {
-            path_to_blockchain: self.path_to_blockchain.lock()?,
-        })
+impl Drop for BlockStore {
+    fn drop(&mut self) {
+        let path = self.path_to_blockchain.join(LOCK_FILE_NAME);
+        let _ = fs::remove_file(path); // we don't care if this succeeds or not
```

Comment:
Shouldn't it at least print a warning/debug message in case of failure?

Reply:
No. The purpose of this PR is to not do that.

Reply:
I don't really understand why it shouldn't be done, but I am fine with that.